### PR TITLE
Bump the minimum version of PostgreSQL to 9.5

### PR DESCRIPTION
**Pull request by yahonda**
This pull request bumps the minimum version of PostgreSQL to 9.6.

#### PostgreSQL 9.3 to 9.5 are EOL
https://www.postgresql.org/support/versioning/

- PostgreSQL 9.3 is EOL since November 2018
- PostgreSQL 9.4 is EOL since February 2020
- PostgreSQL 9.5 is EOL since Feb 11, 2021

#### PostgreSQL 9.4 or higher supports line geometric type

- 8.8. Geometric Types

https://www.postgresql.org/docs/9.4/datatype-geometric.html
`line` geometric type says `Infinite line`, removed `(not fully implemented)`

https://www.postgresql.org/docs/9.3/datatype-geometric.html
`line` geometric type says `Infinite line (not fully implemented)`

### Other Information

Proposed at https://discuss.rubyonrails.org/t/proposal-to-bump-the-minimum-version-of-postgresql-to-9-6-and-mysql-to-5-7-for-rails-7/77234

### CODE REVIEW

File: activerecord/test/cases/adapters/postgresql/uuid_test.rb [hunk 1]
```diff
@@ -234,7 +223,7 @@ class UUID < ActiveRecord::Base
       t.uuid "other_uuid_2", default: "my_uuid_generator()"
     end
 
-    connection.create_table("pg_uuids_3", id: :uuid, **uuid_default) do |t|
+    connection.create_table("pg_uuids_3", id: :uuid, **{}) do |t|
```
Comment (kamipo) [inline activerecord/test/cases/adapters/postgresql/uuid_test.rb:226]:
```suggestion
    connection.create_table("pg_uuids_3", id: :uuid) do |t|
```

File: activerecord/test/cases/adapters/postgresql/uuid_test.rb [hunk 2]
```diff
@@ -376,10 +361,10 @@ class UuidComment < ActiveRecord::Base
 
   setup do
     connection.transaction do
-      connection.create_table("pg_uuid_posts", id: :uuid, **uuid_default) do |t|
+      connection.create_table("pg_uuid_posts", id: :uuid, **{}) do |t|
```
Comment (kamipo) [inline activerecord/test/cases/adapters/postgresql/uuid_test.rb:364]:
```suggestion
      connection.create_table("pg_uuid_posts", id: :uuid) do |t|
```

File: activerecord/test/cases/adapters/postgresql/uuid_test.rb [hunk 3]
```diff
@@ -376,10 +361,10 @@ class UuidComment < ActiveRecord::Base
...
         t.string "title"
       end
-      connection.create_table("pg_uuid_comments", id: :uuid, **uuid_default) do |t|
+      connection.create_table("pg_uuid_comments", id: :uuid, **{}) do |t|
```
Comment (kamipo) [inline activerecord/test/cases/adapters/postgresql/uuid_test.rb:367]:
```suggestion
      connection.create_table("pg_uuid_comments", id: :uuid) do |t|
```

File: activerecord/test/schema/postgresql_specific_schema.rb [hunk 4]
```diff
@@ -2,15 +2,13 @@
 
 ActiveRecord::Schema.define do
   enable_extension!("uuid-ossp", ActiveRecord::Base.connection)
-  enable_extension!("pgcrypto",  ActiveRecord::Base.connection) if ActiveRecord::Base.connection.supports_pgcrypto_uuid?
+  enable_extension!("pgcrypto",  ActiveRecord::Base.connection)
 
-  uuid_default = connection.supports_pgcrypto_uuid? ? {} : { default: "uuid_generate_v4()" }
-
-  create_table :uuid_parents, id: :uuid, force: true, **uuid_default do |t|
+  create_table :uuid_parents, id: :uuid, force: true, **{} do |t|
```
Comment (kamipo) [inline activerecord/test/schema/postgresql_specific_schema.rb:7]:
```suggestion
  create_table :uuid_parents, id: :uuid, force: true do |t|
```

File: activerecord/test/schema/postgresql_specific_schema.rb [hunk 5]
```diff
@@ -2,15 +2,13 @@
...
     t.string :name
   end
 
-  create_table :uuid_children, id: :uuid, force: true, **uuid_default do |t|
+  create_table :uuid_children, id: :uuid, force: true, **{} do |t|
```
Comment (kamipo) [inline activerecord/test/schema/postgresql_specific_schema.rb:11]:
```suggestion
  create_table :uuid_children, id: :uuid, force: true do |t|
```

File: activerecord/test/schema/postgresql_specific_schema.rb [hunk 6]
```diff
@@ -106,23 +104,23 @@
   end
 
   create_table :uuid_comments, force: true, id: false do |t|
-    t.uuid :uuid, primary_key: true, **uuid_default
+    t.uuid :uuid, primary_key: true, **{}
```
Comment (kamipo) [inline activerecord/test/schema/postgresql_specific_schema.rb:107]:
```suggestion
    t.uuid :uuid, primary_key: true
```

File: activerecord/test/schema/postgresql_specific_schema.rb [hunk 7]
```diff
@@ -106,23 +104,23 @@
...
     t.string :content
   end
 
   create_table :uuid_entries, force: true, id: false do |t|
-    t.uuid :uuid, primary_key: true, **uuid_default
+    t.uuid :uuid, primary_key: true, **{}
```
Comment (kamipo) [inline activerecord/test/schema/postgresql_specific_schema.rb:112]:
```suggestion
    t.uuid :uuid, primary_key: true
```

File: activerecord/test/schema/postgresql_specific_schema.rb [hunk 8]
```diff
@@ -106,23 +104,23 @@
...
     t.string :entryable_type, null: false
     t.uuid :entryable_uuid, null: false
   end
 
   create_table :uuid_items, force: true, id: false do |t|
-    t.uuid :uuid, primary_key: true, **uuid_default
+    t.uuid :uuid, primary_key: true, **{}
```
Comment (kamipo) [inline activerecord/test/schema/postgresql_specific_schema.rb:118]:
```suggestion
    t.uuid :uuid, primary_key: true
```

File: activerecord/test/schema/postgresql_specific_schema.rb [hunk 9]
```diff
@@ -106,23 +104,23 @@
...
     t.string :title
   end
 
   create_table :uuid_messages, force: true, id: false do |t|
-    t.uuid :uuid, primary_key: true, **uuid_default
+    t.uuid :uuid, primary_key: true, **{}
```
Comment (kamipo) [inline activerecord/test/schema/postgresql_specific_schema.rb:123]:
```suggestion
    t.uuid :uuid, primary_key: true
```

**Comment by yahonda** [comment]
Thanks for the review. Add commits and squashed.

**Comment by yahonda** [comment]
[CI failure against PostgreSQL with Ruby master](https://buildkite.com/rails/rails/builds/75313#7ce8978a-4795-4bae-af61-c56eb21608e8/1061-1072) is not relevant to this pull request itself. Opened #41528

**Comment by matthewd** [comment]
My first thought is maybe we should go for 9.5... it's old, and I don't feel any particular attachment to supporting it, but it also feels a bit arbitrary to me for us to refuse to connect to a server that speaks the protocol we require, _just_ because it's old. 

(Contrast dropping support for 9.3, which allows us to clean up the UUID stuff... or dropping 9.4, which is more minor, but does remove a conditional.)

Unlike Ruby, where we partly choose to be aggressive in our support policy to help keep the community moving forward, I don't think we have the standing or the need to take that role in the DBMS ecosystem. (And in orgs where the database version lags that much, it's probably outside the control of the application developers to fix: instead of dragging them forward on database versions, we're holding them back on Rails versions.)

Basically, when people ask "why don't you support $oldpostgres?", I prefer to be able to answer "supporting that adds extra code we'd have to maintain" instead of "that's old, you shouldn't be using it". 🤷🏻‍♂️

**Comment by yahonda** [comment]
Thanks for the feedback. I was and am still thinking dropping EOL version of software is appropriate. But I am fine changing this pull request to bump the minimum versions of PostgreSQL 9.5.

**Comment by yahonda** [comment]
Updated pull request version to bump the minimum version of PostgreSQL to 9.5.

File: guides/source/command_line.md [hunk 10]
```diff
@@ -694,7 +694,7 @@ We had to create the **gitapp** directory and initialize an empty git repository
 
 ```bash
 $ cat config/database.yml
-# PostgreSQL. Versions 9.3 and up are supported.
+# PostgreSQL. Versions 9.6 and up are supported.
```
Comment (robin850) [inline guides/source/command_line.md:697]: Last remain from the 9.6 -> 9.5 bump.

Comment (yahonda) [reply guides/source/command_line.md]: Fixed. Thanks for the review.

File: activerecord/CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,7 @@
+* Bump minimum PostgreSQL version to 9.5.
+
+  *Yasuo Honda*
```
Comment (kamipo) [inline activerecord/CHANGELOG.md:3]:
```suggestion
*   Bump minimum PostgreSQL version to 9.5.

    *Yasuo Honda*
```

**Comment by yahonda** [comment]
Thanks for the review. Updated the changelog entry and squashed.
